LearningMissionSimulation/PlaygroundEdvard.cs

```diff
@@ -107,5 +115,25 @@ public void ActivateAccounts()
 
             Console.WriteLine("-----Unfair coordinator finished activating accounts-----");
         }
+
+        public void CreateSubjects(int subjectCount)
+        {
+            for (int i = 0; i < subjectCount; i++)
+            {
+                Subject subject = ObjectGenerator.GenerateSubject();
+
+                subjectDictionary.Add(subject.Id, subject);
+            }
+        }
+
+        public void CreateModules(int moduleCount)
+        {
+            for (int i = 0; i < moduleCount; i++)
+            {
+                Module module = ObjectGenerator.GenerateModule(Guid.NewGuid());
```

Comment:
GenerateModule should take in one of the subject ids that you generated in line 123. Your implementation uses a newly generated Guid, that has NO connection to the generated above subjects. And this is NOT what I want to see. This is a critical point of the implementation and the main requirement. Please think how you can choose randomly an existing subject id. Think about a data structure that can hold your subject ids and that will allow you to pick one of them RANDOMLY every time you need to generate a module.